janusgraph-core/src/main/java/org/janusgraph/core/attribute/Geoshape.java

```diff
@@ -584,23 +595,32 @@ public void write(Kryo kryo, Output output, Geoshape geoshape) {
 
         @Override
         public Geoshape read(Kryo kryo, Input input, Class<Geoshape> aClass) {
-            int length = input.read();
-            assert length>0;
-            InputStream inputStream = new ByteArrayInputStream(input.readBytes(length));
+            final long l = input.readLong();
+            assert l>0 && l<Integer.MAX_VALUE;
+            final int length = (int) l;
             try {
+                final InputStream inputStream = new ByteArrayInputStream(input.readBytes(length));
                 return GeoshapeBinarySerializer.read(inputStream);
-            } catch (IOException e) {
-                throw new RuntimeException("I/O exception reding geoshape");
+            } catch (IOException | KryoException e) {
+                // retry using legacy point deserialization
+                try {
+                    input.setPosition(0);
+                    input.readLong();
+                    return point(input.readFloat(), input.readFloat());
+                } catch (KryoException e2) {
+                    // throw original exception
+                }
+                throw new RuntimeException("I/O exception reading geoshape");
```

Comment:
should you wrap the initial exception e here? `throw new RuntimeException("I/O exception reading geoshape", e);`